src/main/java/org/eclipsefoundation/marketplace/service/CachingService.java

```diff
@@ -0,0 +1,26 @@
+/* Copyright (c) 2019 Eclipse Foundation and others.
+ * This program and the accompanying materials are made available
+ * under the terms of the Eclipse Public License 2.0
+ * which is available at http://www.eclipse.org/legal/epl-v20.html,
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipsefoundation.marketplace.service;
+
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Interface defining the caching service to be used within the application.
+ * 
+ * @author Martin Lowe
+ */
+public interface CachingService {
```

Comment:
What about parameterizing this service (so that `get(String)` returns a `Optional<T>` instead of `Optional<Object>`. It means that a new service will be created by resource kind, but I don't see that as an issue. WDYT?

Also, I would delete `CacheableResource` type and modify the get method to take the 2 params from `CacheableResource::getCacheKey(String, QueryParams)`. As such, the computation of the key is specific to the caching service, and not to the resources. It also avoid having to introduce a super class to resources to support caching.

Reply:
I was actually thinking about that, and was thinking of making one central service mapped by type, but I can parameterize the type on the class instead sure. The only thing I'm worried about is getting items from different caches will be messier, as we will need multiple service references.

I agree with your second point, that is a better idea.

Reply:
> The only thing I'm worried about is getting items from different caches will be messier, as we will need multiple service references.

It should not be too messy. The injection framework should handle that transparently.